python/tests/xir/test_integration.py

```diff
@@ -55,11 +56,14 @@ def parse_script(circuit: str) -> XIRProgram:
 g_one(pi) | [0, 1];
 g_two | [2];
 g_three(1, 3.3) | [2];
+adjoint g_four(1.23) | [2];
+ctrl[0, 2] g_five(3.21) | [1];
 
 // The circuit and statistics
 ry(1.23) | [0];
 rot(0.1, 0.2, 0.3) | [1];
 h(0.2) | [0, 1, 2];
+adjoint ctrl[3] h(0.2) | [0, 1, 2];
```

Comment:
What do you think about moving `adjoint` after `ctrl`?  This may better reflect how the inversion is carried out.
```suggestion
ctrl[3] adjoint h(0.2) | [0, 1, 2];
```

Reply:
This will currently fail since it'll be serialized with `adjoint` first (even though both syntaxes work). I could change the order of the serialization though, if you think it's preferable having `ctrl` come first.

Reply:
Yep, a change would have to be made to the `Statement` serialization; I just thought this would be the clearest place to bring up the point.